lib/pages/my_games_page.dart

```diff
@@ -1,25 +1,80 @@
+import 'package:join_play/utilities/firebase_service.dart';
 
-import '../navigation/route_names.dart';
+class MyGamesPage extends StatefulWidget {
+  final FirebaseService firebaseService;
+  final AuthenticationBloc authenticationBloc;
 
-class MyGamesPage extends StatelessWidget {
-  const MyGamesPage({super.key});
+  const MyGamesPage({
+    super.key,
+    required this.firebaseService,
+    required this.authenticationBloc,
+  });
 
+  @override
+  State<MyGamesPage> createState() => _MyGamesPageState();
+}
+
+class _MyGamesPageState extends State<MyGamesPage> {
   @override
   Widget build(BuildContext context) {
+    final userId = widget.authenticationBloc.sportUser?.uuid ?? '';
+
     return Scaffold(
       appBar: AppBar(
         title: const Text("My Games"),
       ),
-      body: Center(
-        child: FilledButton(
-          child: const Text("Go to History"),
-          onPressed: () {
-            context.goNamed(RouteNames.history);
-          },
-        ),
+      body: FutureBuilder<List<SportEvent>>(
+        future: widget.firebaseService.getUserRegisteredEvents(userId, true), // Fetch future events
+        builder: (context, snapshot) {
+          if (snapshot.connectionState == ConnectionState.waiting) {
+            return const Center(child: CircularProgressIndicator());
+          } else if (snapshot.hasError) {
+            return Center(child: Text("Error: ${snapshot.error}"));
+          } else if (!snapshot.hasData || snapshot.data!.isEmpty) {
+            return const Center(child: Text("No upcoming events found."));
+          } else {
+            final events = snapshot.data!;
+            return ListView.builder(
+              itemCount: events.length,
+              itemBuilder: (context, index) {
+                final event = events[index];
+                return Card(
+                  margin: const EdgeInsets.all(8.0),
+                  child: ListTile(
+                    title: Text(event.name ?? ''),
+                    subtitle: Text(
+                      "Location: ${event.location}\n"
+                      "Time: ${event.dateTime?.toDate()}\n"
+                      "Slots Available: ${event.slotsAvailable}",
+                    ),
+                    trailing: ElevatedButton(
+                      onPressed: () async {
+                        await widget.firebaseService.unregisterFromEvent(
+                          event.id!,
+                          userId,
+                        );
+                        ScaffoldMessenger.of(context).showSnackBar(
+                          SnackBar(content: Text("Unregistered from ${event.name}")),
+                        );
+
+                        // Refresh the UI by calling setState
+                        setState(() {});
+                      },
+                      style: ElevatedButton.styleFrom(
+                        backgroundColor: Colors.red,
+                      ),
```

Comment:
Use theme. 

```suggestion
                      style: ElevatedButton.styleFrom(
                        backgroundColor: Theme.of(context).textTheme.error,
                      ),
```

You might need to add the import statement